pageserver/src/tenant.rs

```diff
@@ -3399,7 +3399,11 @@ impl Tenant {
         // is in progress (which is not a common case).
         //
         // See more for on the issue #2748 condenced out of the initial PR review.
-        let mut shared_cache = self.cached_logical_sizes.lock().await;
+        let mut shared_cache = tokio::select! {
+            locked = self.cached_logical_sizes.lock() => locked,
+            _ = cancel.cancelled() => anyhow::bail!("cancelled"),
+            _ = self.cancel.cancelled() => anyhow::bail!("tenant is shutting down"),
+        };
```

Comment:
I like this change but the PR description doesn't mention it. Would be cleaner to have separate PRs for behavioral change and "just metrics".

Reply:
I am not exactly sure if this fixes anything, I just added it in a whim because I was going for a hotfix and it cannot hurt.

This is likely unnecessary because everything in synthetic size calculation is fast OR cancellation-sensitive already (logical sizes, update gc info).